include/swift/AST/DiagnosticsSema.def

```diff
@@ -3050,6 +3050,22 @@ ERROR(derivative_attr_original_already_has_derivative,none,
       "a derivative already exists for %0", (DeclName))
 NOTE(derivative_attr_duplicate_note,none,
      "other attribute declared here", ())
+ERROR(derivative_attr_access_level_lower_than_original,none,
```

Comment:
Currently, there are two diagnostics regarding original/derivative function access level mismatch:
1. If original is more visible than derivative, blame derivative for not having same visibility.
2. If derivative is more visible than original, blame original for not having same visibility.
```
(1) error: derivative function for public original function 'public_original_private_derivative' must also be public, but '_public_original_private_derivative' is fileprivate
@derivative(of: public_original_private_derivative)
                ^

(2) error: the original function of a public derivative function must also be public, but 'internal_original_public_derivative' is internal
@derivative(of: internal_original_public_derivative)
                ^
```

I think this "placement of blame" is quite weird. What might be better is:
- A blame-free "original access level does not match derivative access level" error.
- Two fixits for either adjusting the original's accessl level or the derivative's.

Any thoughts? @rxwei

Reply:
My preference is always blaming the derivative.

Reply:
I like that and will look into it! The `@derivative` attribute is the trigger, which justifies always blaming the derivative.

Reply:
Done. The new diagnostics look like:
```console
err.swift:21:17: error: derivative function must have same access level as original function; derivative function '_usablefrominline_original_public_derivative' is public, but original function 'usablefrominline_original_public_derivative' is internal
@derivative(of: usablefrominline_original_public_derivative)
                ^
err.swift:22:13: note: mark the derivative function as 'internal' to match the original function
public func _usablefrominline_original_public_derivative(_ x: Float) -> (value: Float, pullback: (Float) -> Float) {
~~~~~~      ^
internal
```